### PR TITLE
Recruiter view refactor

### PR DESCRIPTION
**Pull request by mntss**

### CODE REVIEW

File: src/main/java/serializers/SlotsSerializer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import domain.Slots;
+import java.io.IOException;
+
+/**
+ * Created by jakubr on 09/08/16.
+ */
+public class SlotsSerializer extends StdSerializer<Slots>{
+
+    public SlotsSerializer(){
+        this(null);
```
Comment (wojciechh-kainos) [inline src/main/java/serializers/SlotsSerializer.java]: Passing nulls in java is dangerous

File: src/main/java/serializers/SlotsSerializer.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import domain.Slots;
+import java.io.IOException;
+
+/**
+ * Created by jakubr on 09/08/16.
```
Comment (wojciechh-kainos) [inline src/main/java/serializers/SlotsSerializer.java]: Let's not pollute the files with intellij comments - can we remove these?

File: src/main/java/serializers/SlotsSerializer.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this(null);
+    }
+
+    public SlotsSerializer(Class<Slots> t){
+        super(t);
+    }
+
+    @Override
+    public void serialize(Slots slot, JsonGenerator jgen, SerializerProvider provider) throws IOException, JsonProcessingException{
+        jgen.writeStartObject();
```
Comment (wojciechh-kainos) [inline src/main/java/serializers/SlotsSerializer.java:18]: Is there a test in place that will fail when the structure of the object changes?

File: src/main/resources/assets/js/application/recruiter/services/tdprPersonsService.js [hunk 4]
```diff
@@ -1,20 +1,11 @@
 define(['application/recruiter/tdprRecruiterModule'], function (tdprRecruiterModule) {
     tdprRecruiterModule.service('tdprPersonsService', function ($http, dateFilter, $q) {
         var persons;
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/recruiter/services/tdprPersonsService.js]: The service should be stateless - if we want to preserve persons data - use resolve

File: src/main/resources/assets/js/application/recruiter/services/tdprSlotsTimesService.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+define(['angular', 'application/recruiter/tdprRecruiterModule'], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.service('tdprSlotsTimesService', ['$http', '$q', function ($http, $q) {
+
+        var slotsTimes = {};
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/recruiter/services/tdprSlotsTimesService.js]: A service should be stateless - to preserve the data, use resolve

File: src/main/resources/assets/js/application/recruiter/views/tdpr-recruiter-table.html [hunk 6]
```diff
@@ -1,8 +1,34 @@
...
+        <table id="availabilities-table">
+            <thead>
+                <tr>
+                    <td></td>
+                    <td class="day-th" ng-repeat="day in days">{{day | date: 'EEEE dd.MM.yyyy'}}</td>
+                </tr>
+                <tr>
+                    <td></td>
+                    <td ng-repeat="day in days">
+                        <table style="width: 100%;">
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/recruiter/views/tdpr-recruiter-table.html]: Move inline styles to css files.

File: src/main/resources/assets/js/test/personsServiceTest.js [hunk 7]
```diff
@@ -27,8 +27,8 @@ define(['angular', 'angularMocks', 'application/recruiter/services/tdprPersonsSe
             it('should call backend with date from current week', function () {
                 $httpBackend.expectGET('api/person/all?startDate=' + weekStart + '&endDate=' + weekEnd).respond(200);
 
-                service.fetchPersons().then(function (response) {
-                    expect(response.status).toEqual(200);
+                service.fetchPersonsWithSlotsForDates(weekStart, weekEnd).then(function (response) {
+                expect(response.status).toEqual(200);
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/test/personsServiceTest.js]: Did you check that these tests fail when the response is not 200?

File: src/main/resources/assets/js/application/recruiter/directives/tdprSlotDirective.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+            restrict: 'AE',
+            templateUrl: 'js/application/recruiter/views/tdpr-directive-slot.html',
+            replace: true,
+            scope: {
+                slotData: '='
+
+            },
+            link: function (scope, element, attributes) {
+                scope.getClass = function () {
+                  return scope.slotData ? 'slot-' + scope.slotData.type : '';
```
Comment (drauf) [inline src/main/resources/assets/js/application/recruiter/directives/tdprSlotDirective.js]: you probably should use classes from the AvailabilityEnum

Comment (kwidzinskidawid) [inline src/main/resources/assets/js/application/recruiter/directives/tdprSlotDirective.js]: its safer to get it from the data we recieve from server. enum on front end is not needed

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js [hunk 9]
```diff
@@ -1,30 +1,12 @@
...
-    tdprRecruiterModule.controller("tdprWeekTableController", function ($scope, tdprRecruiterGetSlotsTimesService, tdprPersonsService, JobProfileEnum) {
-        var slotsData = tdprRecruiterGetSlotsTimesService.getSlotsTimes();
-        var personsData = tdprPersonsService.fetchPersons();
+    ], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.controller("tdprWeekTableController", function ($scope, tdprPersonsService, tdprDateService, persons, slotsTimes) {
 
-        $scope.staticData = [];
-        $scope.timeData = {};
-        $scope.jobProfile = JobProfileEnum.dev;
+        $scope.currentJobProfile = "isDev";
```
Comment (jraniszewski) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js]: We should use enum type.

File: src/main/resources/assets/js/application/recruiter/views/tdpr-recruiter-table.html [hunk 10]
```diff
@@ -1,8 +1,49 @@
-<h1>TIMETABLE</h1>
+<div class="container">
 
-<div filter-directive></div>
+    <h1 id="recruiter-header">Availabilities for interviews</h1>
 
-<div data-my-timetable data-start-week-day="startDateWeek" data-model-input="staticData" data-slots-times="timeData" data-selected-day="selectedDay"></div>
+    <div class="legend">
+        <label class="mouse-pointer">
+            <input type="radio" ng-model="currentJobProfile" value="isDev"/>
```
Comment (jraniszewski) [inline src/main/resources/assets/js/application/recruiter/views/tdpr-recruiter-table.html]: Can we replace value with enum value?

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js (hunk 9 above)
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js]: Yup, use enum value here

File: src/main/resources/assets/js/test/personsServiceTest.js (hunk 7 above)
Comment (mntss) [inline src/main/resources/assets/js/test/personsServiceTest.js]: Test with response status 400 added

**Pull request merged**